src/app/loadout-builder/process.ts

```diff
@@ -416,13 +418,32 @@ export function generateMixesFromPerks(
   return mixes;
 }
 
+function getPlugHashesFromCategory(category: DimSocketCategory) {
+  return category.sockets
+    .map((socket) => {
+      return idx(socket, (socket) => socket.plug.plugItem.hash) || null;
+    })
+    .filter(Boolean);
+}
+
 function getBaseStatValue(stat: DimStat, item: DimItem) {
   let baseStatValue = stat.value;
 
   // Checking energy tells us if it is Armour 2.0
   if (item.isDestiny2() && item.sockets && item.energy) {
+    const masterworkSocketCategory = item.sockets.categories.find((category) => {
+      return category.category.categoryStyle === DestinySocketCategoryStyle.EnergyMeter;
+    });
+    const masterworkSocketHashes =
+      (masterworkSocketCategory && getPlugHashesFromCategory(masterworkSocketCategory)) || [];
```

Comment:
One thing to consider is that this code will run for every armor item, for every stat on that armor item. It might be possible to optimize this.

Reply:
Yeah agreed, I didn't think about it too much as I was planning on doing a little bit of an overhaul with some optimisations but I will do it now.

On a side note the live stats are still not being calculated for class items after the update so I might try and fix it here.